packages/core/src/odata-common/request-builder/get-by-key-request-builder.ts

```diff
@@ -68,19 +69,33 @@ export abstract class GetByKeyRequestBuilder<
     destination: Destination | DestinationNameAndJwt,
     options?: DestinationOptions
   ): Promise<EntityT> {
-    return this.build(destination, options)
-      .then(request => request.execute())
-      .then(response =>
+    return this.executeRaw(destination, options)
+      .then(requestResponse =>
         this.entityDeserializer.deserializeEntity(
-          this.dataAccessor.getSingleResult(response.data),
+          this.dataAccessor.getSingleResult(requestResponse.response.data),
           this._entityConstructor,
-          response.headers
+          requestResponse.response.headers
         )
       )
       .catch(error => {
         throw new ErrorWithCause('OData get by key request failed!', error);
       });
   }
+
+  /**
+   * Execute request and return the request and the raw response.
+   *
+   * @param destination - Destination to execute the request against
+   * @param options - Options to employ when fetching destinations
+   * @returns A promise resolving to an [[HttpRequestAndResponse]].
+   */
+  async executeRaw(
```

Comment:
Can you check if you can move this method in the `MethodRequestBuilder` to have higher reuse. The build method is already there.

Reply:
Currently, there is a technical difficulty, where update request builder v2 needs a check for additional warning messages, which is injected between the two steps of the `executeRaw` below:
- building the final request
- execute it
See comment in the `packages/core/src/odata-v2/request-builder/update-request-builder.ts`